Added Docs For Updating to a New Java Version in CI (#166159)
flutter/flutter: docs/platforms/android/Uploading-New-Java-Version-to-CIPD.md

# Uploading New Java Version to CIPD

> We store the Java Version package on CIPD because we want to efficiently store and use
> the binaries for testing. For more information read the internal

Reviewer: ```suggestion
We store the Java Version package on CIPD for use on CI. For more information read the internal
```

I think anything beyond what it is used for gets into a description of what CIPD itself is for, which is out of scope.

Author: done

> docs [here](go/flutter-luci-playbook#remove-duplicated-cipd-tags).

Reviewer: I don't think this link provides docs that relate to the above sentence?

Author: Ah wrong link. updated it. done.

Some links in the instructions below are Google-internal.

## Steps

### Request Access to CIPD

1. Request temporary write access to upload packages to CIPD
   via http://go/flutter-luci-cipd#requesting-write-read-access-to-cipd-packages.

2. Wait about 5-30 minutes for access rights to sync. To check if your rights have synced, see if
   you are a member of this
   group [here](https://chrome-infra-auth.appspot.com/auth/groups/google%2Fflutter-cipd-writers@twosync.google.com).

### Download the New Java Version and Set Up for CIPD Upload

3. Download the new Java version "compressed

>    archive" [here](https://www.oracle.com/java/technologies/downloads) for the following platforms:

Reviewer: OpenJDK used to be different from Oracle's Java, and I assume we were/are using OpenJDK for licensing/legal reasons. Is this no longer the case, or was that assumption wrong?

Author: Oh, I didn't know that. I had assumed they were all fair game. I'm gonna re-upload using OpenJDK. I updated the docs accordingly.

Reviewer: Resolving as docs change looks good. Let me know if there are issues using an open jdk version

# Uploading New Java Version to CIPD

We store the Java Version package on CIPD for use on CI. For more information read the internal

> docs [here](https://g3doc.corp.google.com/company/teams/chrome/ops/luci/cipd.md?cl=head).

Reviewer: Is there a go link we can provide? I don't think we should link directly to g3docs

Author: done. Also a linter may be good for this?
